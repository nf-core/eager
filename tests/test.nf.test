nextflow_pipeline {

    name "Test pipeline: NFCORE_EAGER"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_eager"
    tag "test"

    test("test_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot( path("$outputDir/bam_filtering")            ).match("bam_filtering") },
                { assert snapshot( path("$outputDir/bam_input_stats")          ).match("bam_input_stats") },
                { assert snapshot( path("$outputDir/deduplication")            ).match("deduplication") },
                { assert snapshot( path("$outputDir/mapping")                  ).match("mapping") },
                { assert snapshot( path("$outputDir/samtools")                 ).match("samtools") }, // not sure what this is...
                /*
                    Exceptions (variable md5sums)
                */

                // Damage Estimation
                { assert snapshot( getAllFilesFromDir("$outputDir/damage_estimation/"    , '.txt' )  ).match("damage_estimation_txt" )  },
                { assert snapshot( getAllFilesFromDir("$outputDir/damage_estimation/"    , '.json')  ).match("damage_estimation_json") },
                { assert snapshot( getAllFileNamesFromDir("$outputDir/damage_estimation/", '.svg' )  ).match("damage_estimation_svg" ) },
                { assert snapshot( getAllFileNamesFromDir("$outputDir/damage_estimation/", '.pdf' )  ).match("damage_estimation_pdf" ) },
                { assert snapshot( checkAllFilesNamesFromDirForString("$outputDir/damage_estimation/", '.log', 'FINISHED SUCCESSFULLY')  ).match("damage_estimation_log") },

                // Preprcoessing

                { assert snapshot( path("$outputDir/preprocessing/fastp")          ).match("processing_fastp") }, // TODO: Stable: JSON, varaible: HTML, LOG
                { assert snapshot( path("$outputDir/preprocessing/fastp")          ).match("processing_fastp") }, // TODO: Stable: HTML, variable: ZIP
                //{ assert snapshot( path("$outputDir/preprocessing/fastqc_raw").list().findAll{ it.toString() }).match("processing_fastqc_raw") },
                //{ assert snapshot( path("$outputDir/preprocessing/fastqc_preprocessed").list().findAll{ it.toString() }).match("processing_fastqc_preprocessed") },
                //{ assert snapshot( path("$outputDir/final_bams")               ).match("final_bams") },
                //{ assert snapshot( path("$outputDir/mapstats")                 ).match("mapstats") },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/multiqc/multiqc_data/").exists() }
            )
        }
    }
}

// Recursively list all files in a directory and its sub-directories, matching a given suffix
// TODO: use regex pattern instead of suffix?
def getAllFilesFromDir(dir, suffix) {
    def output = []
    new File(dir).eachFileRecurse() {
        if (it.name.toString().endsWith(suffix)) {
            output.add(it)
        }
    }
    return output.sort()
}
// Recursively list all files names in a directory and its sub-directories, matching a given suffix, return file names
def getAllFileNamesFromDir(dir, suffix) {
    def output = []
    new File(dir).eachFileRecurse() {
        if (it.name.toString().endsWith(suffix)) {
            output.add(it.toString().split("/")[-1])
        }
    }
    return output.sort()
}

// Recursively list all files names in a directory and its sub-directories, matching a given suffix, return if check if given string is in file
def checkAllFilesNamesFromDirForString(dir, suffix, string) {
    def output = []
    new File(dir).eachFileRecurse() {
        if (it.name.toString().endsWith(suffix)) {
            output.add(it.text.contains(string))
        }
    }
    return output.sort()
}
