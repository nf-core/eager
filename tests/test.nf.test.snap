{
    "damage_estimation_txt": {
        "content": [
            [
                "3pGtoA_freq.txt:md5,c32ef382b55fed96816df63993c520eb",
                "3p_freq_misincorporations.txt:md5,2d4d9e2906c2238d27fa7e441bba2b33",
                "5pCtoT_freq.txt:md5,d6e1401fb49387f83d842977ee47a9d2",
                "5p_freq_misincorporations.txt:md5,af2192b8a97699519d060baa3c8982fe",
                "DNA_comp_genome.txt:md5,0f27b55f4a7c2731f35761be4feddbfa",
                "DNA_composition_sample.txt:md5,1336769b56151d9de542610c4a8ef662",
                "editDistance.txt:md5,7c00cd7cc4bd87c2da4debaa2704a66a",
                "lgdistribution.txt:md5,3c176293ea8e58d1d56a95e693727f7a",
                "misincorporation.txt:md5,e25101ff61a0a3c377f8bc0252ee4451",
                "3pGtoA_freq.txt:md5,59aa1691a1c420dffeff4edc4a9cb0dd",
                "3p_freq_misincorporations.txt:md5,15f64d35fc97bab2fed172fec292c3c9",
                "5pCtoT_freq.txt:md5,51511bb6d9278afeb35099ab8b117119",
                "5p_freq_misincorporations.txt:md5,fc0939e3e176deef644c0bd24cb0d141",
                "DNA_comp_genome.txt:md5,76762da07599aff410c2ec531cd19720",
                "DNA_composition_sample.txt:md5,0a4d26c613bbc4541eb51ec1344d6ef4",
                "editDistance.txt:md5,5feeaa4b7018320fd9937e271299375c",
                "lgdistribution.txt:md5,3728eba42dd0b577a8566c103c342eb4",
                "misincorporation.txt:md5,a2d9fed1148fe02e372f0905c7e15b40",
                "3pGtoA_freq.txt:md5,24ae86cb4dd4e558412344a98d582bae",
                "3p_freq_misincorporations.txt:md5,fe312a520a01f198bf248f396b179aff",
                "5pCtoT_freq.txt:md5,5a04f777d38930e205213298e70e09f1",
                "5p_freq_misincorporations.txt:md5,8ecb71fca75a2f86cf44eb0cf4bfd269",
                "DNA_comp_genome.txt:md5,b78d2826f0e1be4990120105c83d0b06",
                "DNA_composition_sample.txt:md5,dd78c33e4d9367759e82c31247685a3b",
                "editDistance.txt:md5,059c352bf858c5ddb27e88253654e353",
                "lgdistribution.txt:md5,5e8760bf271b743aa3c7c96414fb9991",
                "misincorporation.txt:md5,983d8fab464d67481b222aff57b06587"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-22T08:49:40.907112716"
    },
    "damage_estimation_pdf": {
        "content": [
            [
                "edit_distance.pdf",
                "Length_plot.pdf",
                "DamagePlot.pdf",
                "edit_distance.pdf",
                "Length_plot.pdf",
                "DamagePlot.pdf",
                "edit_distance.pdf",
                "Length_plot.pdf",
                "DamagePlot.pdf"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-22T08:49:41.006776905"
    },
    "samtools": {
        "content": [
            [
                "JK2782_JK2782_TGGCCGATCAACGA_BAM.bam.bai:md5,159e97ddc9270c8d941f42f5553cf4fe"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-28T13:08:37.202608535"
    },
    "bam_input_stats": {
        "content": [
            [
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_8_Mammoth_MT_Krause.flagstat:md5,9a10991f5670a52975a622a8b056e2bc"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-28T11:25:56.650730893"
    },
    "bam_filtering": {
        "content": [
            [
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_filtered.bam.bai:md5,d1823ee5a37ad32007d0923731127935",
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_filtered.flagstat:md5,39fd430cc77fc71f71d75a221c1b7880",
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_lengthonly.filtered.bam.bai:md5,159e97ddc9270c8d941f42f5553cf4fe",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_filtered.bam.bai:md5,3829853588ac224ac094d93054c7fbef",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_filtered.flagstat:md5,a9fbd8c9a6f8bb02b191f4140f91d04a",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_lengthonly.filtered.bam.bai:md5,cea7aec832f94e0221bb7da743f3e16c",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_filtered.bam.bai:md5,8f0a7ad0e353b6714701a6c5ae54d3f1",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_filtered.flagstat:md5,f6c739b729c06e17ca205c64478ad823",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_lengthonly.filtered.bam.bai:md5,43a0d7aaeb7c8dae073c829504a4278f"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-28T11:25:56.604142252"
    },
    "damage_estimation_json": {
        "content": [
            [
                "dmgprof.json:md5,e829fb37f3e12842a6139e1d4cfe19dc",
                "dmgprof.json:md5,49a1a9b445abf48904ead652a16fb7e4",
                "dmgprof.json:md5,98c584c20659b11692cbd1e280deeb97"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-22T08:49:40.952167687"
    },
    "deduplication": {
        "content": [
            [
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_dedupped.bam:md5,cafd8aec87c34a34a599aaf881bac4c2",
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_dedupped.bam.bai:md5,196d7c4df7b32afc5209fdf7fff96a37",
                "JK2782_JK2782_TGGCCGATCAACGA_BAM_Mammoth_MT_Krause_dedupped.flagstat:md5,715d840d3bab5a05934825cd9b799ef3",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_dedupped.bam:md5,5d76afc0e40def51f3078613b1f1940f",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_dedupped.bam.bai:md5,9efc80f161fda84978104195171fe4d3",
                "JK2782_JK2782_TGGCCGATCAACGA_Mammoth_MT_Krause_dedupped.flagstat:md5,dde5f59b006afdae2ff5fab85bc8ea92",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_dedupped.bam:md5,8f105b750d30fe800171fd1b6253796a",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_dedupped.bam.bai:md5,51dc155d22f4996a7fe3abb8094767a0",
                "JK2802_JK2802_AGAATAACCTACCA_Mammoth_MT_Krause_dedupped.flagstat:md5,fb4d234136f242c20db27979caa604cf"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-28T11:25:56.733997039"
    },
    "damage_estimation_svg": {
        "content": [
            [
                "DamagePlot_three_prime.svg",
                "Length_plot_combined_data.svg",
                "edit_distance.svg",
                "DamagePlot_five_prime.svg",
                "Length_plot_forward_reverse_separated.svg",
                "DamagePlot_three_prime.svg",
                "Length_plot_combined_data.svg",
                "edit_distance.svg",
                "DamagePlot_five_prime.svg",
                "Length_plot_forward_reverse_separated.svg",
                "DamagePlot_three_prime.svg",
                "Length_plot_combined_data.svg",
                "edit_distance.svg",
                "DamagePlot_five_prime.svg",
                "Length_plot_forward_reverse_separated.svg"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-22T08:49:40.983721036"
    },
    "damage_estimation_log": {
        "content": [
            [
                true,
                true,
                true
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-19T16:38:34.86128972"
    }
}